### ISSUE TITLE
`test-hash` seems to be failing on CentOS

### ISSUE THREAD

**Issue by evverx** [systemd/systemd#8278]
```sh
$ cat /proc/version
Linux version 3.10.0-693.17.1.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-16) (GCC) ) #1 SMP Thu Jan 25 20:13:58 UTC 2018

$ ./build/test-hash
Assertion 'khash_new(&h, "sha256") >= 0' failed at ../src/test/test-hash.c:45, function main(). Aborting.
Aborted

$ strace -s500 -yy -v ./build/test-hash
...
socket(AF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0) = 3<ALG:[226043]>
bind(3<ALG:[226043]>, {sa_family=AF_ALG, sa_data="hash\0\0\0\0\0\0\0\0\0\0"}, 88) = -1 ENOENT (No such file or directory)
close(3<ALG:[226043]>)                  = 0
socket(AF_ALG, SOCK_SEQPACKET|SOCK_CLOEXEC, 0) = 3<ALG:[226047]>
bind(3<ALG:[226047]>, {sa_family=AF_ALG, sa_data="hash\0\0\0\0\0\0\0\0\0\0"}, 88) = 0
mmap(0x601e00000000, 65536, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x601e00000000
accept4(3<ALG:[226047]>, NULL, NULL, SOCK_CLOEXEC) = 4<ALG:[226051]>
sendto(4<ALG:[226051]>, NULL, 0, 0, NULL, 0) = -1 ENOKEY (Required key not available)
recvfrom(4<ALG:[226051]>, 0x601e0000ef60, 129, 0, NULL, NULL) = -1 ENOKEY (Required key not available)
close(3<ALG:[226047]>)                  = 0
close(4<ALG:[226051]>)                  = 0
mmap(0x607200000000, 131072, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x607200000000
writev(2</dev/pts/0>, [{"Assertion 'khash_new(&h, \"sha256\") >= 0' failed at ../src/test/test-hash.c:45, function main(). Aborting.", 105}, {"\n", 1}], 2Assertion 'khash_new(&h, "sha256") >= 0' failed at ../src/test/test-hash.c:45, function main(). Aborting.
) = 106
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
tgkill(12624, 12624, SIGABRT)           = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=12624, si_uid=1000} ---
+++ killed by SIGABRT +++
```
The kernel configuration can be found at https://git.centos.org/blob/rpms!kernel.git/6202b1a49906b21f0cdce08b4ff853a65c658750/SOURCES!kernel-3.10.0-x86_64.config.

**Comment by poettering** [comment]
This sounds very much like a kernel bug. A backport gone wrong or so. ENOKEY suggests that the kernel appears to think that sha256 was a keyed hash function, which it is not. Can you contact the CentOS kernel maintainers on that?

I am a bit reluctant to change systemd to convert ENOKEY to EOPNOTSUPP in this case silently, since ENOKEY is a valid reply on keyed hash functions when no key is passed...

**Comment by poettering** [comment]
So, I am pretty sure this is a kernel problem and should be fixed there, but i prepped fix in #8282 anyway, given that this kind of kernel is probably going to stay with us for a while

**Comment by poettering** [comment]
@evverx would be great if you could tets #8282 on your centos system!
